lib/spack/spack/cmd/env.py

```diff
@@ -446,6 +448,40 @@ def env_deactivate(args):
     sys.stdout.write(cmds)
 
 
+#
+# env track
+#
+def env_track_setup_parser(subparser):
+    """track an environment from a directory in Spack"""
+    subparser.add_argument("-n", "--name", help="custom environment name")
+    subparser.add_argument("dir", help="path to environment")
+    arguments.add_common_arguments(subparser, ["yes_to_all"])
+
+
+def env_track(args):
```

Comment:
I think there should be a `spack env untrack` command that is essentially the same as `spack env rm` EXCEPT:

- `untrack` should only remove a tracked environment (not any other env)
- it should be an error to `untrack` a normally managed environment (basically so the user has a safe way to indicate they intend only to untrack, not to blow away an entire existing env)
- `rm` should untack a tracked env OR remove a managed env, and say which it did